packages/orm/src/EntityManager.ts

```diff
@@ -1063,10 +1063,13 @@ export class EntityManager<C = unknown> {
     this.#pendingCascadeDeletes.push(entity);
   }
 
-  async assignNewIds() {
+  assignNewIds() {
     let pendingEntities = this.entities.filter((e) => e.isNewEntity && !e.isDeletedEntity && !e.idMaybe);
-    let todos = createTodos(pendingEntities);
-    return this.driver.assignNewIds(this, todos);
+    return this.getLoader<Entity, void>("assign-new-ids", "global", async (entities) => {
+      let todos = createTodos(Array.from(entities));
+      await this.driver.assignNewIds(this, todos);
+      return [];
+    }).loadMany(pendingEntities);
```

Comment:
I'm only on mobile so I'm not 100% sure but my guess is we might need to pass primitives like maybe pendingEntities.map(e => e.toString())?

We'd want that to be something stable, which I think toString would be...

Reply:
No style guide, but I think we usually do '|| fail(...)' to avoid the bangs.

Mapping back to entities makes sense. Fwiw maybe do a lookup in a map to avoid the scans that could be a N^2.

I think we have a groupBy utility method for making the list into a map.

Reply:
Oh yeah, instead of passing IDs plus look it up in a map, we can probably pass a custom hash function to the getLoader function ... I don't have a link handy, but some of our other getLoader calls do that, like in the findDataLoader iirc

Reply:
I think due to the nature of the em, a cache key isn't actually required. Each entity is only going to exist once and internally Dataloader uses a Map<>. Happy to use `cacheKeyFn` though if desired

Reply:
Ah sure, that makes sense!